### ISSUE TITLE
Add support for dark syntax highlighting

### ISSUE THREAD

**Issue by Akkadius** [squidfunk/mkdocs-material#768]
Not so much an issue as it is for anyone else who finds it useful

Saw a few other threads that posted about this but not one that had it 100% - made a few tweaks on my own including the clipboard highlighter

Make sure in your mkdocs.yml you have the following

```
markdown_extensions:
  - codehilite:
      guess_lang: false
      use_pygments: true
      noclasses: true
      pygments_style: monokai
```

And then use an extra.css file as such

```
extra_css:
  - css/extra.css
```

In that file use

```css
pre { color: white !important; }

.md-clipboard:before {
    color: rgb(255, 255, 255);
}

.codehilite:hover .md-clipboard:before,.md-typeset .highlight:hover .md-clipboard:before,pre:hover .md-clipboard:before {
    color: rgba(255, 255, 255, 0.54) !important
}
```

Yields

![image](https://user-images.githubusercontent.com/3319450/39166160-dce0ccec-474d-11e8-8197-2000d5dc0212.png)

**Comment by squidfunk** [comment]
Thanks for your example! I want to support dark code highlighting within Material but it's a really low priority issue. I will leave this issue open for directions and as a feature request, so that we implement it at some point in the future.

**Comment by squidfunk** [comment]
I just experimented with the Pygments themes and they work for code blocks, but not for inline code snippets. Pygments doesn't apply any style for inline `<code>` sections. This means that changing the style will only change code blocks. I was hoping to add your hack to the documentation as a guide but I'm not quite satisfied with the result because a lot of custom tweaking is necessary per syntax theme, see:

<img width="671" alt="bildschirmfoto 2018-04-29 um 17 12 16" src="https://user-images.githubusercontent.com/932156/39408047-823b974a-4bd0-11e8-8b6b-f65f55569e8b.png">

@facelessuser - have you experimented with light/dark themes? Does the Highlight extension provided by Pymdown solve this problem?

**Comment by facelessuser** [comment]
@squidfunk, if you use [pymdownx.inlinehilite](https://facelessuser.github.io/pymdown-extensions/extensions/inlinehilite/) you can use a special syntax that forces inline blocks to go through pygments to be highlighted. If you have language guessing enabled, then I think it might highlight inline blocks that it identifies as a language, but I don't remember if I allow that.

But if you want all inline blocks that don't use the syntax to go through pygments (language blocks defined as plain text or that default to plain text), you can just enable the option [style_plain_text](https://facelessuser.github.io/pymdown-extensions/extensions/inlinehilite/#options) and all inline code blocks will be run through pygments.

So in short, InlineHilite specifically allows this.  The the `Highlight` extension is not currently required for this functionality.

**Comment by facelessuser** [comment]
Correction, `InlineHilite` runs any inline block that as the special language specifier through pygments.    `style_plain_text` will additionally run inline blocks without a language specifier through pygments.  If language guessing is enabled, it will run all through blocks through pygments as blocks without language specifiers are run through the language guesser.

**Comment by squidfunk** [comment]
This issue actually provides a way to enable dark syntax highlighting which is achievable in less than 5 minutes. Implementing a default within Material will probably make no sense as everybody likes different colors, so for this reason I decided to leave the colors as-is. Just following the steps above and it should work out of the box with all themes supported by Pygments.
